core/src/test/scala/unit/kafka/admin/ListConsumerGroupTest.scala

```diff
@@ -111,37 +263,124 @@ class ListConsumerGroupTest extends ConsumerGroupCommandTest {
+      out = TestUtils.grabConsoleOutput(ConsumerGroupCommand.main(cgcArgs))
+      out.contains("TYPE") && !out.contains("STATE") && out.contains(simpleGroup) && out.contains(group) && out.contains(protocolGroup)
+    }, s"Expected to find $simpleGroup, $group, $protocolGroup and the header, but found $out")
+
+    cgcArgs = Array("--bootstrap-server", bootstrapServers(), "--list", "--state", "--type")
+    TestUtils.waitUntilTrue(() => {
+      out = TestUtils.grabConsoleOutput(ConsumerGroupCommand.main(cgcArgs))
+      out.contains("TYPE") && out.contains("STATE") && out.contains(simpleGroup) && out.contains(group) && out.contains(protocolGroup)
+    }, s"Expected to find $simpleGroup, $group, $protocolGroup and the header, but found $out")
 
     cgcArgs = Array("--bootstrap-server", bootstrapServers(), "--list", "--state", "Stable")
     TestUtils.waitUntilTrue(() => {
       out = TestUtils.grabConsoleOutput(ConsumerGroupCommand.main(cgcArgs))
-      out.contains("STATE") && out.contains(group) && out.contains("Stable")
-    }, s"Expected to find $group in state Stable and the header, but found $out")
+      out.contains("STATE") && out.contains(group) && out.contains("Stable") && out.contains(protocolGroup)
+    }, s"Expected to find $group, $protocolGroup in state Stable and the header, but found $out")
 
     cgcArgs = Array("--bootstrap-server", bootstrapServers(), "--list", "--state", "stable")
     TestUtils.waitUntilTrue(() => {
       out = TestUtils.grabConsoleOutput(ConsumerGroupCommand.main(cgcArgs))
-      out.contains("STATE") && out.contains(group) && out.contains("Stable")
-    }, s"Expected to find $group in state Stable and the header, but found $out")
+      out.contains("STATE") && out.contains(group) && out.contains("Stable") && out.contains(protocolGroup)
+    }, s"Expected to find $group, $protocolGroup in state Stable and the header, but found $out")
+
+    cgcArgs = Array("--bootstrap-server", bootstrapServers(), "--list", "--type", "Classic")
+    TestUtils.waitUntilTrue(() => {
+      out = TestUtils.grabConsoleOutput(ConsumerGroupCommand.main(cgcArgs))
+      out.contains("TYPE") && out.contains("Classic") && !out.contains("STATE") && out.contains(simpleGroup) && out.contains(group) && out.contains(protocolGroup)
+    }, s"Expected to find $simpleGroup, $group, $protocolGroup and the header, but found $out")
+
+    cgcArgs = Array("--bootstrap-server", bootstrapServers(), "--list", "--type", "classic")
+    TestUtils.waitUntilTrue(() => {
+      out = TestUtils.grabConsoleOutput(ConsumerGroupCommand.main(cgcArgs))
+      out.contains("TYPE") && out.contains("Classic") && !out.contains("STATE") && out.contains(simpleGroup) && out.contains(group) && out.contains(protocolGroup)
+    }, s"Expected to find $simpleGroup, $group, $protocolGroup  and the header, but found $out")
   }
 
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumAndGroupProtocolNames)
+  @CsvSource(Array("kraft+kip848,consumer"))
```

Comment:
It would be better to also have a method as the source (like the others) for this one. We can add one if does not exist yet.

Reply:
Oh yes I considered that but since it was just one case I thought we could just pass it as a csv, I can add a method